public/video-ui/src/services/UploadsApi.js

```diff
@@ -31,10 +26,27 @@ class UploadFunctions {
     });
   };
 
-  stopUpload = id => {
-    return pandaReqwest({
-      url: `/api2/uploads/${id}`,
-      method: 'delete'
+  uploadParts = (upload, parts, file, progressFn) => {
+    return new Promise((resolve, reject) => {
+      const uploadPartRecursive = parts => {
+        if (parts.length === 0) {
+          resolve(true);
+        } else {
+          const part = parts[0];
+
+          this.uploadPart(upload, part, file, progressFn)
+            .then(s3Request => {
+              return s3Request.promise().then(() => {
+                uploadPartRecursive(parts.slice(1));
```

Comment:
minor: make this `resolve(uploadPartRecursive(parts.slice(1))` to be more explicit in the `Promise` you're returning